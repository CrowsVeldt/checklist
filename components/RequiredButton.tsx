import { AntDesign } from "@expo/vector-icons";
import { useState } from "react";
import { Pressable, StyleSheet, Text } from "react-native";

export default function RequiredButton({
  initialState,
  changeState,
}: {
  initialState: boolean;
  changeState: any;
}) {
  const [active, setActive] = useState<boolean>(initialState);
  return (
    <Pressable
      style={({ pressed }) =>
        pressed ? [styles.required, styles.pressed] : styles.required
      }
      onPress={() => {
        setActive(!active);
        changeState(!active);
      }}
    >
      <AntDesign
        name="exclamationcircleo"
        size={24}
        color={active ? "black" : "whitesmoke"}
      />
    </Pressable>
  );
}

const styles = StyleSheet.create({
  required: {
    justifyContent: "center",
    alignItems: "center",
    height: "100%",
    width: 50,
    borderStartColor: "black",
    borderStartWidth: 1,
  },
  pressed: {
    backgroundColor: "lightgray",
  },
});
