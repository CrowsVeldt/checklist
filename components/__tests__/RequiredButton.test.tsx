import { render } from "@testing-library/react-native";
import RequiredButton from "../RequiredButton";

describe("<RequiredButton />", () => {
  test("RequiredButton renders correctly", () => {
    const tree = render(
      <RequiredButton initialState={false} changeState={undefined} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
