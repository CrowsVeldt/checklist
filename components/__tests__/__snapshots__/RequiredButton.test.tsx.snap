// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequiredButton /> RequiredButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderStartColor": "black",
      "borderStartWidth": 1,
      "height": "100%",
      "justifyContent": "center",
      "width": 50,
    }
  }
>
  <Text />
  <View
    style={
      {
        "borderColor": "gray",
        "borderWidth": 1,
        "position": "absolute",
        "transform": [
          {
            "rotate": "45deg",
          },
        ],
        "width": "100%",
      }
    }
  />
</View>
`;
